### PR TITLE
Add type definition for cartesian conversion methods

### PR DESCRIPTION
**Pull request by HoldYourWaffle**
Methods were originally added by #14684 but the type definitions were never updated, so I figured I'd do it myself

### CODE REVIEW

File: src/math/Spherical.d.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export class Spherical {
 	clone(): this;
 	copy( other: Spherical ): this;
 	makeSafe(): void;
-	setFromVector3( vec3: Vector3 ): Spherical;
+	setFromVector3( vec3: Vector3 ): this;
```
Comment (Mugen87) [inline src/math/Spherical.d.ts]: You can also replace `Spherical` with `this` for the `set()` method.

File: src/math/Vector3.d.ts [hunk 2]
```diff
@@ -236,7 +236,9 @@ export class Vector3 implements Vector {
 	distanceToManhattan( v: Vector3 ): number;
 
 	setFromSpherical( s: Spherical ): this;
+	setFromSphericalCoords( r: number, phi: number, theta:number ): this;
 	setFromCylindrical( s: Cylindrical ): this;
+	setFromCylindricalCoords( r: number, theta: number, y: number ): this;
```
Comment (Mugen87) [inline src/math/Vector3.d.ts]: I would rename `r` to `radius`, like in the JS code.

**Comment by HoldYourWaffle** [comment]
I have addressed your concerns and squashed my 'fixup' commits

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**